### ISSUE TITLE
Feature request: Forcejoin and Forceleave commands.

### ISSUE THREAD

**Issue by GrammaticaDW** [garbagemule/MobArena#594]
# Feature request

**Short description**
If a player (or console) execute /ma forcejoin %player% %arena% or /ma forceleave %player% %arena%. Then the specific player will join of will leave the specific arena 

**Implementation details**
The specific player joins the specific arena. No configurations needed.

**Additional info**
I need this command because my players are abusing the /ma join command. Thx for reading.

**Comment by garbagemule** [comment]
Isn't this easily solved by using a sudo plugin?

**Comment by GrammaticaDW** [comment]
No, I have quests with bossfights on my server. Now a player can easily join a arena without doing any quest. And then the player can farm the rewards.

**Comment by garbagemule** [comment]
I'm pretty stupid when it comes to server setups, so I think I'm going to need some more context.

I don't understand how a potential force join/leave mechanism would be relevant to questing. Can you explain your problem and why you believe a force join/leave mechanism is the right solution? There's not enough information here for me to understand why you need this, and I'm not going to add it "just because", since:
- A `sudo` style plugin can run `/ma join` or `/ma leave` as if it was the player. I don't see how this is any different. It's just introducing a new command that does something for the player instead of the player doing it themselves, and this is already possible with other plugins.
- A `cron` style plugin can enable and disable the arenas or the plugin as a whole on a schedule. This is one way to limit access to MobArena if that's the problem. See the discussion in issue #534 about why "fun limiting" rubs me the wrong way.

**Comment by Nesseley** [comment]
I can see how it would benefit a quest system to have a admin-command to send people in and out of arenas.

If you can use /sudo, the player itself can also just use /ma join whenever, which is bad if using the plugin as a quest/boss thingy as stated they will be able to start and farm quest items etc. Enabling and disabling the plugin is also a way to do it, but that could mess up for other players in a quest etc.

**The way i would do it:**
Have the arena in another world.
Use max-join-distance setting for the arena.
When the player needs to join the arena, make the quest-plugin warp them to the arena-world and sudo join them.
After arena is finished sudo the player out and back to where he was before.

**Comment by GrammaticaDW** [comment]
Nessely is right. Quests are a huge part of a roleplay server. We use MobArena for our bosses in combination with quests 🙂

Our player can obtain legendary items by defeating certain kind of bosses.
At this moment every player can join with  /ma join %arena% and join every arena they like.
With this command every player can join the Arena over and over again en they will get rewards every time. If they do this the player can farm resources and that's not the thing we want. Because we don't want players to join every Arena with a simple command.

We use a custom GUI to join arenas. At the moment an icon in our GUI's is connected with the command /sudo %player% ma arena %boss%
I would like to change this to /ma forcejoin %player% %arena%
So I can disable the permission so that players cant use /ma join. So they can't join via command but only with the GUI

**Comment by garbagemule** [comment]
Can the GUI grant permission temporarily just to join, then remove it again afterwards?

**Comment by Sketaful** [comment]
Sorry for just jumping in, but...
There is plugins that let you run commands as if the player used it (suod) but as OP, so if you remove the permission to join arenas this should still work. 
You could also add the permission to the player to be able to join an arena, sudo them to join and then remove the permission.

There should be a bunch of ways to do this apart from the ones above. I understand how a forcejoin would be easier, but it isn't a must?
